hack/rp-dev/automate_rp_deploy.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash -e
+######## Helper file to automate the full RP dev creation ########
+
+echo "##### Make sure to be logged in to Azure prior to running this script ####"
+echo "##### In case of failure when creating Azure reseource, consider running the rp-full-dev-clenup target ####"
+echo "#### E.g., AZURE_PREFIX=$AZURE_PREFIX LOCATION=eastus make rp-full-dev-clenup       "
+source hack/rp-dev/rp_funcs.sh 
+setup_rp_config $AZURE_PREFIX $SKIP_DEPLOYMENTS
+pre_deploy_resources
+add_hive
+mirror_images
+prepare_RP_deployment
```

Comment:
These functions can potentially run in parallel to speed up the process as they are independent IMHO

Reply:
Thanks @s-fairchild! My comment was more of a side note on whether you agree on running them in parallel. 
Regarding implementation, your suggestion seems very nice

Reply:
Yes, we should run these in parallel to save time.

Reply:
I didn't try to support concurrency but I would prefer to address that in a follow-up. Reviewing the PR is already quite complex.